src/test/java/org/apache/drillui/test/framework/pages/QueryResultsPage.java

```diff
@@ -37,6 +39,106 @@
   @FindBy(xpath = "//*[@title=\"Open in new window\"]")
   private WebElement profileButton;
 
+  @FindBy(xpath = "/html/body/div[2]/div[2]/table/tbody/tr/td[1]/button/span[1]")
+  private WebElement queryResultStatus;
+
+  @FindBy(id = "delimitBy")
+  private WebElement exportCSVDelimiter;
+
+  @FindBy(xpath = "/html/body/div[2]/div[2]/table/tbody/tr/td[3]/button")
+  private WebElement exportButton;
+
+  @FindBy(xpath = "//*[@id=\"result_filter\"]/label/input")
+  private WebElement rowsFilter;
+
+  @FindBy(xpath = "//*[@id=\"result_wrapper\"]/div[1]/div[2]/button")
+  private WebElement showHideColumns;
+
+  @FindBy(xpath = "/html/body/ul")
+  private WebElement columnsListFilter;
+
+  @FindBy(className = "ColVis_collectionBackground")
+  private WebElement outsideColumnsFilter;
+
+  @FindBy(xpath = "//*[@id=\"result_length\"]/label/select")
+  private WebElement rowsPerPage;
+
+  @FindBy(id = "result_info")
+  private WebElement pageRowsInfo;
+
+  @FindBy(id = "result_previous")
+  private WebElement previousPage;
+
+  @FindBy(xpath = "//*[@id=\"result_paginate\"]/span")
+  private WebElement pagesList;
+
+  @FindBy(id = "result_next")
+  private WebElement nextPage;
+
+  public boolean hasPrevPage() {
+    return !previousPage.getAttribute("class").endsWith("ui-state-disabled");
+  }
+
+  public boolean hasPaginationPages() {
+    return pagesList.findElements(By.tagName("a")).size() > 1;
+  }
+
+  public int getPaginationPagesCount() {
+    return pagesList.findElements(By.tagName("a")).size();
+  }
+
+  public void openPage(int pageNumber) {
+    pagesList.findElements(By.tagName("a")).get(pageNumber - 1).click();
+  }
+
+  public boolean hasNextPage() {
+    return !nextPage.getAttribute("class").endsWith("ui-state-disabled");
+  }
+
+  public String getQueryStatus() {
+    return queryResultStatus.getText();
+  }
+
+  public String getPageRowsInfo() {
+    return pageRowsInfo.getText();
+  }
+
+  public void findInRows(String text) {
+    rowsFilter.clear();
+    rowsFilter.sendKeys(text);
+  }
+
+  public void filterColumns(List<String> columns) {
+    showHideColumns.click();
+    List<WebElement> columnsElements = new LinkedList<>();
+    columnsElements.addAll(columnsListFilter.findElements(By.tagName("label")));
```

Comment:
You should temporary set implicit timeout to 0 before findElements() method call, or otherwise it would wait the whole timeout time.